### PR TITLE
Fix paper-slider overflow issue

### PR DESCRIPTION
**Pull request by cdce8p**
My attempt at fixing #1219. I wasn't able to define a style override, which is why I defined `ha-paper-dialog-scrollable`. If someone with more experience knows how to do it, I'll update this PR.

So far I've tried adding this to the style section of `MoreInfoControls`.
```js
      :host([domain=cover]) paper-dialog-scrollable {
        --paper-dialog-scrollable {
          overflow: visible;
        }
      }
```

----

The solution would like like this and will work with `camera` more-info cards as well.

fixed

### CODE REVIEW

File: src/components/ha-paper-dialog-scrollable.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const PaperDialogScrollable = customElements.get('paper-dialog-scrollable');
+
+class HaPaperDialogScrollable extends PaperDialogScrollable {
+  static get template() {
+    const tpl = document.createElement('template');
+    tpl.innerHTML = PaperDialogScrollable.template.innerHTML;
+    const styleEl = tpl.content.querySelector('style');
+    styleEl.setAttribute('include', 'paper-dialog-scrollable');
+    styleEl.innerHTML += `
```
[the remark this answers is not in the document]
Comment (cdce8p) [reply src/components/ha-paper-dialog-scrollable.js]: That's what I though, too. Unfortunately
```js
styleEl.setAttribute('include', 'paper-dialog-scrollable');
```
seems to have no effect, although `ha-paper-slider` uses it as well.

**Comment by balloob** [comment]
How does this impact the scrollable element when it is required to scroll?

**Comment by balloob** [comment]
This breaks the component. Tried it by adding `overflow: visible` to the component on the demo page.

Normal:
image

With this fix:
image

**Comment by cdce8p** [comment]
In which cases to we use the scrolling? I haven't come across any yet.

**Comment by balloob** [comment]
On mobile phones when the more info dialog is very tall.

I wonder if this can help? https://css-tricks.com/popping-hidden-overflow/

**Comment by cdce8p** [comment]
I've looked into it. Sadly it breaks the paper-dialog-scrollable as well.
I'm not sure anymore there is an easy fix for it.

broken

**Comment by cdce8p** [comment]
What if we were to remove the `pin` altogether?
That's how it's done for `light/brightness` and other as well.

**Comment by balloob** [comment]
Well the pin is nice because you can see the value while you are moving you finger. We actually made the pin bigger so it worked better with fingers. We can also make it smaller again 😉

**Comment by cdce8p** [comment]
The pins are now smaller. In addition I added a small padding around `Position` and `Tilt position`.

unbenannt

File: src/dialogs/more-info/controls/more-info-cover.js [hunk 2]
```diff
@@ -23,11 +23,18 @@ import featureClassNames from '../../../common/entity/feature_class_names';
         max-height: 0px;
         overflow: hidden;
       }
+
+      .current_position > div,
+      .tilt > .name {
+        padding-top: 4px;
```
Comment (balloob) [inline src/dialogs/more-info/controls/more-info-cover.js]: This feels weird. We should try to have a single styling for all our sliders in more info.

Comment (cdce8p) [reply src/dialogs/more-info/controls/more-info-cover.js]: I'm open to suggestions. My reasoning was just that the slider use cases are already fairly customized and adding the padding to the top of the slider would look wired.

Comment (balloob) [reply src/dialogs/more-info/controls/more-info-cover.js]: We should use the same labeled slider everywhere and that should have the same look https://github.com/home-assistant/home-assistant-polymer/blob/master/src/components/ha-labeled-slider.js

**Comment by cdce8p** [comment]
I updated the design to use `ha-labeled-slider`. Since the `title` already has a bottom margin which is slightly larger than the padding before, I've increase the `pin` size a bit.

new

File: src/components/ha-labeled-slider.js [hunk 3]
```diff
@@ -24,19 +24,20 @@ class HaLabeledSlider extends PolymerElement {
...
       ha-paper-slider {
         background-image: var(--ha-slider-background);
       }
     </style>
 
     <div class="title">[[caption]]</div>
-    <iron-icon icon="[[icon]]"></iron-icon>
+    <template is="dom-if" if="[[extra]]">
+      <div class="extra-container">
+        <slot></slot>
```
Comment (balloob) [inline src/components/ha-labeled-slider.js]: let's give the slot a name. I also don't think that we need a `dom-if` around it. It won't be rendered if it's not there.

File: src/components/ha-labeled-slider.js [hunk 4]
```diff
@@ -24,19 +24,20 @@ class HaLabeledSlider extends PolymerElement {
...
+      </div>
+    </template>
     <div class="slider-container">
-      <ha-paper-slider min="[[min]]" max="[[max]]" value="{{value}}" ignore-bar-touch="[[ignoreBarTouch]]">
+      <iron-icon icon="[[icon]]" hidden\$="[[!icon]]"></iron-icon>
+      <ha-paper-slider min="[[min]]" max="[[max]]" step="[[step]]" pin="[[pin]]" value="{{value}}" disabled="[[disabled]]" ignore-bar-touch="[[ignoreBarTouch]]">
```
Comment (balloob) [inline src/components/ha-labeled-slider.js]: Not a super big fan of long lines, prefer to break up like this:

```html
<ha-paper-slider
  min="[[min]]"
  max="[[max]]"
  …
/></ha-paper-slider>
```

**Comment by cdce8p** [comment]
Updated according to comments. I just didn't put each variable on a separate line, IMO that's too much. Used a readable format instead.

**Review by balloob — APPROVED** [review]

**Pull request merged**